flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerImplTest.java

```diff
@@ -18,166 +18,187 @@
-		Mockito.when(jobVertex.getJobVertexId()).thenReturn(new JobVertexID());
-		Mockito.when(jobVertex.getGraph()).thenReturn(graph);
-		Mockito.when(jobVertex.getTaskVertices()).thenReturn(taskVertices);
-
-		taskVertices[0] = mockExecutionVertex(jobVertex, 0);
-		taskVertices[1] = mockExecutionVertex(jobVertex, 1);
-		taskVertices[2] = mockExecutionVertex(jobVertex, 2);
-		taskVertices[3] = mockExecutionVertex(jobVertex, 3);
-
-		int numSamples = 100;
-		Time delayBetweenSamples = Time.milliseconds(100L);
-
-		BackPressureStatsTrackerImpl tracker = new BackPressureStatsTrackerImpl(
-				sampleCoordinator, 9999, numSamples, Integer.MAX_VALUE, delayBetweenSamples);
-
-		// getOperatorBackPressureStats triggers stack trace sampling
-		Assert.assertFalse(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		Mockito.verify(sampleCoordinator, Mockito.times(1)).triggerStackTraceSample(
-				Matchers.eq(taskVertices),
-				Matchers.eq(numSamples),
-				Matchers.eq(delayBetweenSamples),
-				Matchers.eq(BackPressureStatsTrackerImpl.MAX_STACK_TRACE_DEPTH));
-
-		// Request back pressure stats again. This should not trigger another sample request
-		Assert.assertTrue(!tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		Mockito.verify(sampleCoordinator, Mockito.times(1)).triggerStackTraceSample(
-				Matchers.eq(taskVertices),
-				Matchers.eq(numSamples),
-				Matchers.eq(delayBetweenSamples),
-				Matchers.eq(BackPressureStatsTrackerImpl.MAX_STACK_TRACE_DEPTH));
-
-		Assert.assertTrue(!tracker.getOperatorBackPressureStats(jobVertex).isPresent());
-
-		// Complete the future
-		Map<ExecutionAttemptID, List<StackTraceElement[]>> traces = new HashMap<>();
-		for (ExecutionVertex vertex : taskVertices) {
-			List<StackTraceElement[]> taskTraces = new ArrayList<>();
+	public void testGetOperatorBackPressureStats() throws Exception {
+		doInitialRequestAndVerifyTheResult(createBackPressureTracker(60000, 60000, backPressureStats));
+	}
 
-			for (int i = 0; i < taskVertices.length; i++) {
-				// Traces until sub task index are back pressured
-				taskTraces.add(createStackTrace(i <= vertex.getParallelSubtaskIndex()));
-			}
+	@Test
+	public void testCachedStatsNotUpdatedWithinRefreshInterval() throws Exception {
+		final double backPressureRatio2 = 0.2;
+		final BackPressureStats backPressureStats2 = createBackPressureStats(1, 60000, backPressureRatio2);
+
+		final BackPressureStatsTracker tracker = createBackPressureTracker(60000, 60000, backPressureStats, backPressureStats2);
+		doInitialRequestAndVerifyTheResult(tracker);
+		// verify that no new back pressure request is triggered
+		checkOperatorBackPressureStats(tracker.getOperatorBackPressureStats(executionJobVertex));
+	}
 
-			traces.put(vertex.getCurrentExecutionAttempt().getAttemptId(), taskTraces);
-		}
+	@Test
+	public void testCachedStatsUpdatedAfterRefreshInterval() throws Exception {
+		final int backPressureStatsRefreshInterval = 100;
```

Comment:
We can use smaller time 1 instead 100, also for above similar case

Reply:
I reduce the time to 10ms.